plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -351,6 +359,55 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
         return tables.build();
     }
 
+    @Override
+    public Optional<Iterator<TableCommentMetadata>> streamRelationComments(ConnectorSession session, Optional<String> namespace, Predicate<SchemaTableName> isRedirected)
+    {
+        if (!cacheTableMetadata) {
+            return Optional.empty();
+        }
+
+        return Optional.of(listNamespaces(session, namespace).stream()
+                .flatMap(glueNamespace -> getPaginatedResults(
+                        glueClient::getTables,
+                        new GetTablesRequest().withDatabaseName(glueNamespace),
+                        GetTablesRequest::setNextToken,
+                        GetTablesResult::getNextToken,
+                        stats.getGetTables())
+                        .map(GetTablesResult::getTableList)
+                        .flatMap(List::stream)
+                        .map(table -> {
+                            SchemaTableName tableName = new SchemaTableName(glueNamespace, table.getName());
```

Comment:
What about Delta or Hive tables?

Reply:
good catch, will add a isIcebergTable check, and remove isHiveOrPrestoView check